PYME/Acquire/ui/actionUI.py

```diff
@@ -222,5 +225,57 @@ def OnROIsFromTileviewer(self, event):
         print(rois.shape)
         self._add_ROIs(rois)
         
+class PathMinimizingActionPanel(ActionPanel):
+    def __init__(self, parent, actionManager, scope, sort_epsilon=0.01):
+        """
+        Parameters
+        ----------
+        sort_epsilon: float
+            Relative per-index exit tolerance for two-opt sorting. See traveling_salesperson.two_opt
+        """
+        ActionPanel.__init__(self, parent, actionManager, scope)
+        self._sort_epsilon = sort_epsilon
+
+    def _add_ROIs(self, rois):
```

Comment:
Several points:

- I don't like the idea of over-riding `_add_rois()` as it duplicates logic. Much better to call `_add_rois(sort_rois(rois))` or similar.
- There is far too much logic here in what should be GUI code (what happened to model view separation?). The original `_add_ROIs()` function is already pushing it.
- I don't like the idea of subclassing a GUI class to change the backend logic (this is a really gross design pattern)

Reply:
Medium term I could envisage letting ActionUI have a user selectable sort function. As a short term solution I'll suggest the following:

- give `actionUI` an `roi_sort_fcn` property, which defaults to `roi_sort_fcn = lambda rois : rois`. Might need to optimise the signature here depending on what the sort needs to know, e.g. current position? 

- modify `_add_rois()` so it looks like:
```python
def _add_rois(self, rois):
    rois = self.roi_sort_fcn(rois)
    ### rest of _add_rois here ...
```

Reply:
Longer term, potentially factor some of the rest of `_add_rois()` into the ActionManager, rather than actionUI.  

Note, the travelling salesman `roi_sort_fcn` should ideally be in a separate file to the actionUI (although could potentially be a module-level function for now - this lets us write it in a way which semi-separate from the GUI, and worry about where to put it later).